src/blueprint/from_config.rs

```diff
@@ -89,7 +89,11 @@ where
     let schema = match type_ {
         Some(type_) => {
             if let Some(variants) = type_.variants.clone() {
-                JsonSchema::Enum(variants)
+                let mut map: HashMap<i32, HashSet<String>> = HashMap::new();
+                for (i, variant) in variants.into_iter().enumerate() {
+                    map.entry(i as i32).or_default().insert(variant);
+                }
+                JsonSchema::Enum(map)
```

Comment:
do we need new map? Won't `JsonSchema::Enum(variants)` work fine?

Reply:
The change.in json_schema module wants the argument as hashmap.

Because I am using hashmap for validation of enum type in compare method of json_schema module there.

Reply:
for that you might need to change type of `type_.variants` to HashMap as well because it's not guaranteed that the value number lies in the range of number of variants.  

example:

```
enum Foo {
 BAR = 0;
 BAR1 = 59;
}
```
Is perfectly allowed. But with current implementation I think it will never look for number 59.

Reply:
Yes. Nice catch.
That would be the problem in that case. We need to find some way to capture those as well when reading.
not just variants values.

Reply:
change `type_.variants` to the HashMap instead of BTreeSet

Reply:
ok.

Reply:
need to changes some code to solve
self.variants = Some(variants.union(other).cloned().collect());

Reply:
variants is now  a hasmap right.

Reply:
Like if there is conflicts like same keys when merging hashmap then preferring the right side one.?